src/MobileRedirection.php

```diff
@@ -180,6 +186,20 @@ public function redirect() {
 		}
 	}
 
+	/**
+	 * Add mobile version switcher link/styles when mobile redirection is not enabled.
+	 */
+	public function add_mobile_switcher_link_when_redirection_disabled() {
+		if ( $this->is_mobile_request() ) {
+			add_action( 'wp_head', [ $this, 'add_mobile_version_switcher_styles' ] );
+			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
+			add_filter( 'amp_to_amp_linking_element_excluded', [ $this, 'filter_amp_to_amp_linking_element_excluded' ], 100, 2 );
+			add_filter( 'amp_to_amp_linking_element_query_vars', [ $this, 'filter_amp_to_amp_linking_element_query_vars' ], 10, 2 );
+			add_action( 'amp_post_template_head', [ $this, 'add_mobile_version_switcher_styles' ] ); // For legacy Reader mode theme.
+			add_action( 'amp_post_template_footer', [ $this, 'add_mobile_version_switcher_link' ] ); // For legacy Reader mode theme.
```

Comment:
Since these calls are duplicated in the `redirect` method, how about moving them into private methods that can be called in both places? 

`add_mobile_switcher_head_hooks()`:

https://github.com/ampproject/amp-wp/blob/2ed973b04b0d3c3e395839a115daffc6cef4fb7f/src/MobileRedirection.php#L153-L154

`add_a2a_linking_hooks()`:

https://github.com/ampproject/amp-wp/blob/2ed973b04b0d3c3e395839a115daffc6cef4fb7f/src/MobileRedirection.php#L180-L181

`add_mobile_switcher_footer_hooks()`:

https://github.com/ampproject/amp-wp/blob/2ed973b04b0d3c3e395839a115daffc6cef4fb7f/src/MobileRedirection.php#L184-L185

Then this method can be updated to be just:

```php
if ( $this->is_mobile_request() ) {
  $this->add_mobile_switcher_head_hooks();
  $this->add_a2a_linking_hooks();
  $this->add_mobile_switcher_footer_hooks();
}
```

And likewise change the calls in the `redirect` method.

This will help with maintenance down the road.